x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java

```diff
@@ -49,11 +53,20 @@ protected String getJobId() {
 
         @Override
         protected IterationResult<Integer> doProcess(SearchResponse searchResponse) {
+            awaitForLatch();
             assertThat(step, equalTo(3));
             ++step;
             return new IterationResult<Integer>(Collections.emptyList(), 3, true);
         }
 
+        private void awaitForLatch() {
+            try {
+                latch.await();
```

Comment:
Please put a timeout (10s or whatever you think it's appropriate) to prevent this for hanging up.